app-emulation/cri-tools/cri-tools-1.18.0.ebuild

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit coreos-go bash-completion-r1
+
+COREOS_GO_VERSION="go1.15"
+COREOS_GO_PACKAGE="github.com/kubernetes-sigs/cri-tools"
+COREOS_GO_MOD="vendor"
+
+MY_PV="v${PV/_beta/-beta.}"
+
+EGO_PN="${COREOS_GO_PACKAGE}"
+DESCRIPTION="CLI and validation tools for Kubelet Container Runtime (CRI)"
+HOMEPAGE="https://github.com/kubernetes-sigs/cri-tools"
+SRC_URI="https://github.com/kubernetes-sigs/cri-tools/archive/${MY_PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="Apache-2.0 BSD BSD-2 CC-BY-SA-4.0 ISC MIT MPL-2.0"
+SLOT="0"
+KEYWORDS="amd64 arm64"
+
+S=${WORKDIR}/cri-tools-${PV}
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+
+src_compile() {
+	GO_LDFLAGS="-s -w -extldflags=-Wl,-z,now,-z,relro,-z,defs"
```

Comment:
Upstream's Makefile also specifies `-X $(PROJECT)/pkg/version.Version=$(VERSION)` in LDFLAGS, should we also do it? Also, it specifies `-tags selinux`, which we do not. Is it something we should use too?

In general, why not using upstream's Makefile, but using only the `crictl` target? Maybe with patching the upstream's Makefile to add those linker flags for binary size reduction?

Reply:
> Upstream's Makefile also specifies -X $(PROJECT)/pkg/version.Version=$(VERSION) in LDFLAGS, should we also do it?

Yes, let's do that.

> it specifies -tags selinux, which we do not. Is it something we should use too?

The `selinux` tag is anyway enabled by default in upstream cri-tools.
In theory, we can allow it to be disabled via the USE flag, though that needs a lot of work in eclass.

> In general, why not using upstream's Makefile, but using only the crictl target?

ok, let me see if it is possible.

> Maybe with patching the upstream's Makefile to add those linker flags for binary size reduction?

In theory yes, possible.
Will do that in the future, as it needs a lot of time until the upstream PR could be merged.

Reply:
> In general, why not using upstream's Makefile, but using only the crictl target?

No, it is not possible to use upstream cri-tools' Makefile, because the Makefile does not allow custom variables like `BUILDTAGS` or `GO_LDFLAGS` to be configured.
Unfortunately it is the only way to run `go_build` like that.
Added comments about that.